git-secrets

```diff
@@ -61,11 +62,15 @@ load_patterns() {
 }
 
 load_allowed() {
+  local new_rev="$1"
   git config --get-all secrets.allowed
   local gitallowed="$(git rev-parse --show-toplevel)/.gitallowed"
   if [ -e "$gitallowed" ]; then
     cat $gitallowed | awk 'NF && $1!~/^#/'
   fi
+  if [ -n "${new_rev}" ]; then
+    git show ${new_rev}:.gitallowed 2>/dev/null | awk 'NF && $1!~/^#/'
```

Comment:
All this is doing: `git show ${new_rev}:.gitallowed`, is printing the contents of .gitallowed to stdout. it operates in the same way as: `cat $gitallowed`, except that `git show ${new_rev}:.gitallowed` outputs the contents of `.gitallowed` as seen in `new_rev`. I send stderr to /dev/null to catch cases where there is no `.gitallowed` file. I don't know what the awk stuff does, but it is used on line 67|69, so I assumed it is necessary to format the output of this function in a so that whatever calls it can use it. `cat .gitallowed` formats output in exactly the same way as `git show ${new_rev}:.gitallowed`, so I would make sense that their outputs would need to be filtered in the same way also.

Reply:
Or did you mean add a comment to the file?

Reply:
Yes, I meant a comment in the code to help future maintainers understand what's happening. Thanks!

Reply:
Added comment below